LibraBFT/Impl/Consensus/BlockStorage/BlockStore.agda

```diff
@@ -34,6 +48,44 @@ executeAndInsertBlockM b = do
       put (rmSetBlockStore s bs')
       ok eb
 
+executeAndInsertBlockE bs0 block =
+  maybeS (getBlock (block ^∙ bId) bs0) continue (pure ∘ (bs0 ,_))
+ where
+  continue : Either ErrLog (BlockStore _ × ExecutedBlock)
+  continue =
+    maybeS (bs0 ^∙ bsRoot _) (Left fakeErr) λ bsr →
+    let btRound = bsr ^∙ ebRound in
+    if-dec btRound ≥?ℕ block ^∙ bRound
```

Comment:
@cwjnkins has recently improved support for code closer to the Haskell.  Hopefully he can point out where.

Reply:
Just updating that you could look through the changes in #69, not at the proofs, but there are a few examples of where @cwjnkins has used the nicer syntax support in cases like this.

Reply:
@mark-moir @haroldcarr The changes I made would not help with this one, unfortunately (this is not in `RWST`, and I've only done equality, not less-than / greater-than comparisons)